canvas.go

```diff
@@ -23,6 +21,12 @@ const (
 	Lossy
 )
 
+// PixelsPerMillimeter (PPM) for the resolution of raster images. Higher DPM will result in bigger images.
+type PixelsPerMillimeter float64
```

Comment:
Perhaps `Resolution` with additional methods/constructors, see comment below. Not sure if worth it, would like your input on this.

Reply:
_Previous discussion https://github.com/tdewolff/canvas/pull/25#discussion_r395149583_

I think the name or the type should contain the unit in a lengthy name (`PixelsPerMillimeter` or `DotsPerMillimeter`), so that a look at the `PNGWriter` definition makes it immediately clear, what this variable is about (without having to read any comments).

Currently `rasterizer.PNGWriter(1.0)` is just a shortcut for `rasterizer.PNGWriter(canvas.PixelsPerMillimeter(1.0))` (thanks to go type inference).
If we replace `PixelsPerMillimeter` with `Resolution` (or density), I find that it loses some clarity: `rasterizer.PNGWriter(canvas.Resolution(1.0))` (what does a resolution of `1.0` means?). However `resolution` could be a good fit for the variable name (I previously chose `density`, in reference to https://en.wikipedia.org/wiki/Pixel_density).

Using `DPM` also reduces the readability. I would prefer `DotsPerMillimeter`.

Proposition combining the elements (`resolution` and `DotsPerMillimeter`):
```go
func PNGWriter(resolution canvas.DotsPerMillimeter) canvas.Writer
```
What do you think?

Reply:
The terms `DPI` and `PPI` are well known in industry, also `DPCM` for dots-per-centimeter, so I think `DPM` is not really a stretch that needs a lot of explaination. I agree that `Resolution` is a bad name as it can also refer to the size of the image. I think the following is pretty clear:

https://play.golang.org/p/M6U8Ph6Kadi

Reply:
I agree with you, that we should use `DPCM` instead (since it seems more in line with industry standards).

---

I would prefer having a specific unit as type, since I think that it greatly improves the readability of the code and of the documentation. 

```go
type Density float64

// By looking at the function signature (like in godoc)
// I have no clue what this value means (only that a bigger value is a higher density)
func someWriter(d Density) { 
	fmt.Println(d)
}

type DPCM float64

// By looking at the function signature (like in godoc)
// I know that 72 is in DPCM
func otherWriter(d DPCM) {
	fmt.Println(d)
}
```

In both cases can the user choose to be explicit or not:
```go
func main() {
	someWriter(72)
	otherWriter(72)

	someWriter(72 * DPM)
	otherWriter(DPCM(72))
}
```
https://play.golang.org/p/590ONV5UQjj

Reply:
Not bad, but what if we implement a method such as the following? Then any implementation of `Density` is agnostic to the format it stores the number.

```go
func (d Density) DPMM() float64 {
	return d*1.0
}
```

Reply:
I agree that a `DPMM` method is useful for using the `Density` variable, but I still believe that the documentation and readability would be better with an unambiguous type.

Anyway, I don't want to [bikeshed](https://en.wikipedia.org/wiki/Bikeshedding) too much and since it is your project, I will comply with what you choose between:
- `type Density float64`
- `type DPI float64`
- `type DPMM float64`
- `type DPCM float64` (<- my current preference)

Please, let me know :smiley:

Reply:
I agree, I'd say we stick with the least amount of changes for now and we can always check back later on. Instead of `DPCM` though I suggest `DPMM` because we don't want to be testing all functionality again (test coverage does not cover this) such as for fonts. So that'd be your 3rd option. If you have an idea to make it better let's put that in a separate PR.

Sorry for my absence, I'd like to encourage submissions by responding swiftly but I've been put on an emergency project regarding COVID and I've been working day and night. Anyways, I really appreciate the effort you put into this!

Reply:
Alright, I have changed `resolution canvas.DotsPerMillimeter` to `resolution canvas.DPPM`.

> Sorry for my absence

Don't worry, I agree that the current situation can be quite stressful (and a voluntary project like `canvas` should not add more stress to it). Thanks for letting me know, I will try not to bother you too much